### ISSUE TITLE
Why the rows in train_data should be flattened within the from_batches function?

### ISSUE THREAD

**Issue by shuo-nvidia** [NVIDIA-NeMo/RL#1077]
**Describe the bug**

When running GRPO, the function in `nemo_rl\distributed\batched_data_dict.py` lines 135-137 flattens each row of the tensors in `train_data`. However, the rows passed here are all 1D (with a `seq_len` dimension), it seems flattening has no effect.. 

However, when I passed full logits with shape `(batch_size, seq_len, vocab_size)` and enabling sequence packing, this flattening incorrectly merges the last two dimensions (`seq_len` and `vocab_size`), leading to bugs. Why we need the rows flattened within the from_batches function?

**Comment by snowmanwwg** [comment]
@terrykong is there a reason for this? if not, should @shuo-nvidia  raise a PR to fix it?
